Use an adjusted OpenLayers eslint config

marcjansen opened the pull request:
This PR suggests to lint our source code with an adjusted version of the openlayers eslint configuration.

GeoExt and OpenLayers itself use that particular configuration.

This makes our code easier to comprehend, and also revealed some issues, most of which were tagged with `TODO`.comments.

Most changes should be cosmetic, but the color util got a bigger makeover, but also some tests.

Please review.

PS: I added this in that many commits because…

* this makes it slightly easier to review
* this makes it easier to revert / adjust in case I introduced a bug or sth,
* I get paid by commit :stuck_out_tongue_winking_eye:

dnlkoch commented on src/view/button/Measure.js:
```diff
@@ -408,16 +408,18 @@ Ext.define('BasiGX.view.button.Measure', {
+            btnText = '{textangle}';
+            break;
+        default:
+            break;
```
Add/Replace `"indent": ["error", 2, { "SwitchCase": 1 }]` in `.eslintrc` to ensure "correct" indent.

marcjansen replied:
Done.

dnlkoch commented on src/view/button/Measure.js:
```diff
@@ -460,14 +462,17 @@ Ext.define('BasiGX.view.button.Measure', {
     /**
      * Adds a tooltip on click where a measuring stop occured.
+     *
+     * @param {ol.MapBrowserEvent} evt The event which contains teh coordinate
```
Typo: `teh` --> `the`.

marcjansen replied:
Done.

dnlkoch commented on src/view/button/Measure.js:
```diff
@@ -547,7 +557,7 @@ Ext.define('BasiGX.view.button.Measure', {
      * Handle pointer move by updating and repositioning the dynamic tooltip.
      *
-     * @param {ol.MapBrowserEvent} evt
+     * @param {ol.MapBrowserEvent} evt The Event from the pointermove.
```
Typo: `Event` --> `event`.

marcjansen replied:
Done.

dnlkoch commented on src/view/container/NominatimSearch.js:
```diff
@@ -153,9 +165,16 @@ Ext.define("BasiGX.view.container.NominatimSearch", {
+        /**
+         * Returns an array of styles.
+         *
+         * TODO Are we accessing aprvate property here? `getSource().distance_`
```
Typo: `aprvate` --> `private`.

marcjansen replied:
Done.

dnlkoch commented on src/view/container/NominatimSearch.js:
```diff
@@ -512,43 +543,56 @@ Ext.define("BasiGX.view.container.NominatimSearch", {
     /**
      * Works with extent or geom.
+     *
+     * @param {ol.Extent|ol.geom.SimpleGeometry} extent The extent or geoemetry
```
Typo: `geoemetry` --> `geometry`.

marcjansen replied:
Done.

dnlkoch commented on src/view/container/OverpassSearch.js:
```diff
@@ -164,9 +176,16 @@ Ext.define("BasiGX.view.container.OverpassSearch", {
+        /**
+         * Returns an array of styles.
+         *
+         * TODO Are we accessing aprvate property here? `getSource().distance_`
```
Typo: `aprvate` --> `private`.

marcjansen replied:
Done.

dnlkoch commented on src/view/container/OverpassSearch.js:
```diff
@@ -557,97 +584,116 @@ Ext.define("BasiGX.view.container.OverpassSearch", {
     /**
      * Works with extent or geom.
+     *
+     * @param {ol.Extent|ol.geom.SimpleGeometry} extent The extent or geoemetry
```
Typo: `geoemetry` --> `geometry`

marcjansen replied:
Done.

dnlkoch commented on src/view/container/RedlineStyler.js:
```diff
@@ -786,24 +843,28 @@ Ext.define("BasiGX.view.container.RedlineStyler", {
     },
 
     /**
+     * Gets an array of icon sytle attributes from the form elements to
```
Typo: `sytle` --> `style`

marcjansen replied:
Done.

dnlkoch commented on src/view/container/Redlining.js:
```diff
@@ -138,648 +138,685 @@ Ext.define("BasiGX.view.container.Redlining", {
+    },
+
+    /**
+     * Modify a postits text.
```
Typo: `postits` --> `postit`.

marcjansen replied:
Done.

dnlkoch commented on src/view/container/WfsSearch.js:
```diff
@@ -156,9 +167,16 @@ Ext.define("BasiGX.view.container.WfsSearch", {
+        /**
+         * Returns an array of styles.
+         *
+         * TODO Are we accessing aprvate property here? `getSource().distance_`
```
Typo: `aprvate` --> `private`.

marcjansen replied:
Done.

dnlkoch commented on src/view/container/WfsSearch.js:
```diff
@@ -409,12 +433,13 @@ Ext.define("BasiGX.view.container.WfsSearch", {
     /**
-     *
+     * Issues a `DescribeFeatureType` request, so that we know details about the
+     * Featuretyoe we are working on.
```
Typo: `Featuretyoe` --> `featuretype`.

marcjansen replied:
Done.

dnlkoch commented on src/view/form/Print.js:
```diff
@@ -748,44 +808,51 @@ Ext.define("BasiGX.view.form.Print", {
+            attributeFields = me.getStringField(attributeRec);
+            break;
+        default:
+            break;
```
See above (somewhere :-) ).

marcjansen replied:
Done.

dnlkoch commented on src/view/form/field/GazetteerCombo.js:
```diff
@@ -54,22 +54,23 @@ Ext.define('BasiGX.view.form.field.GazetteerCombo',{
     /**
-     * @param combo
-     * @param newValue
+     * When the combo value changes, do a ne w Gazetteer search.
```
Typo: `ne w` --> `new `.
Typo: `Gazetteer` --> `gazetteer`.

marcjansen replied:
Done.

dnlkoch commented on src/view/form/field/MultiSearchCombo.js:
```diff
@@ -150,7 +150,10 @@ Ext.define('BasiGX.view.form.field.MultiSearchCombo',{
 
-    onBoxReady: function(){
+    /**
+     * Ensures that the value is clearer when one hits escape.
```
Typo: `clearer` --> `cleared`.

marcjansen replied:
Done.

dnlkoch commented on src/view/panel/GraphicPool.js:
```diff
@@ -249,23 +251,31 @@ Ext.define("BasiGX.view.panel.GraphicPool", {
     },
 
     /**
+     * Returns the Ext.data.Model instacne for teh selected picture.
```
Typo: `instacne` --> `instance`.
Typo: `teh` --> `the`.

marcjansen replied:
Done.

dnlkoch commented on src/view/panel/LegendTree.js:
```diff
@@ -218,38 +218,46 @@ Ext.define("BasiGX.view.panel.LegendTree", {
     },
 
     /**
+     * Recursively seacrhes the passed record (or its children) for a
```
Typo: `seacrhes` --> `searches`.

marcjansen replied:
Done.

dnlkoch commented on src/view/panel/LegendTree.js:
```diff
@@ -218,38 +218,46 @@ Ext.define("BasiGX.view.panel.LegendTree", {
      * If a layer is configured with property 'treeColor', the color will
      * get applied here. Folders will inherit the color
+     *
+     * @param {Ext.data.Model} rec The tree node recordof the row.
```
Typo: `recordof` --> `record of`.

marcjansen replied:
Done.

dnlkoch commented:
Thanks @marcjansen for this outstanding work!

marcjansen commented on src/util/Color.js:
```diff
@@ -21,90 +21,140 @@ Ext.define('BasiGX.util.Color', {
          *     // rgba is now: "rgba(255,0,0,0.5)"
+         *
+         * @param {String} hex The hex string to convert.
+         * @param {Number} opacity The opacity to set, should be between `` and
```
Done.

marcjansen commented on src/util/ConfigParser.js:
```diff
@@ -484,21 +512,24 @@ Ext.define('BasiGX.util.ConfigParser', {
+                // as described on https://wiki.intranet.terrestris.de/doku.php
+                // ?id=clustering
+                //
+                // There is currently now way in ol3 to request features on
```
Done.

marcjansen commented on src/util/Map.js:
```diff
@@ -73,36 +74,54 @@ Ext.define('BasiGX.util.Map', {
 
-        getResolution: function (map) {
+        /**
+         * Returns the resolution of the passed maps view.
```
Done.

marcjansen commented on src/util/Map.js:
```diff
@@ -73,36 +74,54 @@ Ext.define('BasiGX.util.Map', {
+        /**
+         * Returns the resolution of the passed maps view.
+         *
+         * @param {ol.Map} map The map to get the resolution of.
```
Done.

marcjansen commented on src/util/Url.js:
```diff
@@ -44,33 +47,44 @@ Ext.define('BasiGX.util.Url', {
-         *   Tribulus
+         * TODO Are we sure we have an application always deployed like it is
+         *      assumed herein?
+         *     'folder').
```
Done.

marcjansen commented on src/ux/ContextSensitiveHelp.js:
```diff
@@ -130,24 +148,29 @@ Ext.define('BasiGX.ux.ContextSensitiveHelp', {
         },
 
         /**
+         * Determines the component at the passed pointb that has help info
```
Done.

marcjansen commented:
Thanks for the tedious review work, @dnlkoch. I have addressed all your comments and will merge once everything is green.

Thanks again for the review!
